doc/source/whatsnew/v1.3.0.rst

```diff
@@ -339,6 +339,7 @@ Groupby/resample/rolling
 - Fixed bug in :meth:`DataFrameGroupBy.sum` and :meth:`SeriesGroupBy.sum` causing loss of precision through using Kahan summation (:issue:`38778`)
 - Fixed bug in :meth:`DataFrameGroupBy.cumsum`, :meth:`SeriesGroupBy.cumsum`, :meth:`DataFrameGroupBy.mean` and :meth:`SeriesGroupBy.mean` causing loss of precision through using Kahan summation (:issue:`38934`)
 - Bug in :meth:`.Resampler.aggregate` and :meth:`DataFrame.transform` raising ``TypeError`` instead of ``SpecificationError`` when missing keys had mixed dtypes (:issue:`39025`)
+- Bug in :meth:`Series.resample` would raise when index consisted of ``NaT`` (:issue:`39227`)
```

Comment:
Probably mention that this is specific to `PeriodIndex`?

Construction with a `DatetimeIndex` in your original example raises before `resampling` so the bug fix shouldnt apply to it

```
In [9]: ser = Series([1, 2], index=DatetimeIndex([NaT, NaT], freq=freq))

ValueError: Neither `start` nor `end` can be NaT

The above exception was the direct cause of the following exception:

ValueError: Inferred frequency None from passed values does not conform to passed frequency D
```

Reply:
Thanks!